core/src/main/java/com/godaddy/commerce/core/Expirable.java

```diff
@@ -0,0 +1,26 @@
+package com.godaddy.commerce.core;
+
+import java.time.Instant;
+import lombok.experimental.Accessors;
+import lombok.Getter;
+
+@Accessors(fluent = true)
+public class Expirable<T> {
+  @Getter
+  private final T item;
+
+  private final Instant expiration;
+
+  public Expirable(T t, long maxAge) {
+    if (t == null) {
+      throw new IllegalArgumentException("Item must not be null.");
+    }
+
+    item = t;
```

Comment:
> [<img alt="szhang1" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.secureserver.net/szhang1) **Authored by [szhang1](https://github.secureserver.net/szhang1)**
_<time datetime="2020-11-16T23:18:24Z" title="Tuesday, November 17th 2020, 12:18:24 am +01:00">Nov 17, 2020</time>_

---


Nit:
```suggestion
  public Expirable(T item, long maxAge) {
    if (item == null) {
      throw new IllegalArgumentException("Item must not be null.");
    }

    this.item = item;
```